### PR TITLE
Fix azidentity/cache CI

### PR DESCRIPTION
**Pull request by chlowell**
This skips testing the Linux persistent cache implementation when it isn't useable, as in our CI where the user who executes tests apparently isn't allowed to write files. Skipping those tests requires setting a low coverage bar. I also added a changelog, readme and license file to get the analyze job passing.

### CODE REVIEW

File: sdk/azidentity/cache/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Azure Identity Cache Module for Go
+
+This module implements a cross-platform persistent token cache for [azidentity][https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity] credentials. See that module's [examples](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#pkg-examples) for sample code showing how to configure persistent caching for a credential.
```
Comment (jhendrixMSFT) [inline sdk/azidentity/cache/README.md:3]: Is the link to `azidentity` supposed to be in parenthesis?

File: sdk/azidentity/cache/linux_test.go [hunk 2]
```diff
@@ -18,9 +18,21 @@ import (
...
 
-var ctx = context.Background()
+var (
+	ctx              = context.Background()
+	storageAvailable = false
+)
+
+func TestMain(m *testing.M) {
+	err := tryKeyring()
+	storageAvailable = err == nil
```
Comment (jhendrixMSFT) [inline sdk/azidentity/cache/linux_test.go:28]: What kind of error is happening in CI?  I wonder if there's a way we can fix this so we don't have to skip coverage.

File: sdk/azidentity/cache/README.md (hunk 1 above)
Comment (chlowell) [reply sdk/azidentity/cache/README.md:3]: Yes! Thanks for spotting this

File: sdk/azidentity/cache/linux_test.go [hunk 3]
```diff
@@ -18,21 +18,19 @@ import (
...
-	ctx              = context.Background()
-	storageAvailable = false
-)
+var ctx = context.Background()
 
 func TestMain(m *testing.M) {
-	err := tryKeyring()
-	storageAvailable = err == nil
+	// ensure the user running tests in CI is able to write cache files
+	if tmpdir, ok := os.LookupEnv("Agent.TempDirectory"); ok {
```
Comment (benbp) [inline sdk/azidentity/cache/linux_test.go:25]: Sorry @chlowell I misled you a bit. In the yaml you can use `$(Agent.TempDirectory)` to set new env variables or pass as a parameter. However in the environment it gets converted to `AGENT_TEMPDIRECTORY`.

File: sdk/azidentity/cache/linux_test.go (hunk 2 above)
Comment (chlowell) [reply sdk/azidentity/cache/linux_test.go]: Turns out there were two issues: a test bug used the wrong path, and the user isn't allowed to search its persistent keyring. I fixed the test bug but the permission issue requires more investigation. I think we should skip the tests in CI for now, to unblock an initial beta release while I figure out whether there's a better way to handle or work around this permission issue.

Comment (jhendrixMSFT) [reply sdk/azidentity/cache/linux_test.go]: Glad you found the root cause.

**Pull request merged**